EIPS/eip-6787.md

```diff
@@ -0,0 +1,93 @@
+
+`function deposit(address token, uint256 amount) external;`
+
+The **deposit** function allows a user to deposit a specified amount of a particular token to the exchange. The *token* parameter specifies the address of the token contract, and the *amount* parameter specifies the amount of the token to be deposited.
+
+### withdraw
+
+`function withdraw(address token, uint256 amount) external;`
+
+The **withdraw** function allows a user to withdraw a specified amount of a particular token from the exchange. The *token* parameter specifies the address of the token contract, and the *amount* parameter specifies the amount of the token to be withdrawn.
+
+### settle
+
+`function settle(address userA, address tokenA, uint256 amountA, address userB, address tokenB, uint256 amountB) external;`
+
+The **settle** function is used to settle transactions between two parties, involving the exchange of assets represented by tokenA and tokenB. The function takes the following parameters:
+
+1. *userA*: the address of the first user involved in the transaction
+2. *tokenA*: the address of the token being traded by user A
+3. *amountA*: the amount of token A being traded by user A
+4. *userB*: the address of the second user involved in the transaction
+5. *tokenB*: the address of the token being traded by user B
+6. *amountB*: the amount of token B being traded by user B
+
+### prepareForceWithdraw
+
+`function prepareForceWithdraw(address token, uint256 amount) external returns (uint256 requestID);`
+
+The assets deposited by users will be stored in the exchange contract's account, and the exchange can achieve real-time 1:1 reserve proof. The **prepareForceWithdraw** function is used for users to initiate a forced withdrawal of a certain amount of a specified token. This function indicates that the user wants to perform a forced withdrawal and can submit the withdrawal after the default timeout period. Within the timeout period, the exchange needs to confirm that the user's order status meets the expected criteria, and forcibly cancel the user's order and settle the trade to avoid malicious attacks by the user. This function takes the following parameters:
+
+1. *token*: the address of the token to be withdrawn
+2. *amount*: the amount of the token to be withdrawn
+
+The function returns a unique *requestID* that can be used to submit the forced withdrawal using the commitForceWithdraw function.
+
+### commitForceWithdraw
+
+`function commitForceWithdraw(uint256 requestID) external;`
+
+The **commitForceWithdraw** function is used to execute a forced withdrawal operation after the conditions are met. The function takes a *requestID* parameter, which specifies the ID of the forced withdrawal request to be executed. The request must have been previously initiated using the prepareForceWithdraw function.
+
+The flow charts for **prepareForceWithdraw** and **commitForceWithdraw** are shown below：
+
+![](../assets/eip-6787/image1.png)
+
+## Rationale
+
```

Comment:
You'll need to include some explanation for any technical choices made within this proposal.